tests/onnxruntime/test_modeling.py

```diff
@@ -186,7 +186,10 @@ def test_load_model_from_empty_cache(self):
         dirpath = os.path.join(default_cache_path, "models--" + self.TINY_ONNX_MODEL_ID.replace("/", "--"))
 
         if os.path.exists(dirpath) and os.path.isdir(dirpath):
-            shutil.rmtree(dirpath)
+            if os.name == "nt":
+                os.system(f"rmdir /S /Q {dirpath}")
+            else:
+                shutil.rmtree(dirpath)
```

Comment:
why do you need this on windows? Also, you might want to consider isolating this code and make a function, to avoid repeating it.

Reply:
it's this error https://github.com/huggingface/optimum/actions/runs/9366908110/job/25785437479#step:7:30120
some refs: https://github.com/python/cpython/issues/107408

Reply:
So if it is a workaround for a python issue, I believe it's even more important to isolate it in a function that should be well explained, so if one day we go back to this code we do not ask ourselves "why it is done this way"?

Reply:
done !